identity-account-storage/src/storage/traits.rs

```diff
@@ -27,45 +28,82 @@ mod storage_sub_trait {
   impl<S: Send + Sync + super::Storage> StorageSendSyncMaybe for S {}
 }
 
-/// An interface for Identity Account storage implementations.
+/// An interface for Account storage implementations.
 ///
-/// See [MemStore][crate::storage::MemStore] for a test/example implementation.
+/// The [`Storage`] interface is used for secure operations on keys, such as generation and signing,
+/// as well as key-value like storage of data structures, such as DID documents.
+///
+/// This interface works with [`AccountId`] and [`IotaDID`] as the top-level identifiers.
+/// `AccountId` is intended as a partition-key, i.e. everything related to an account
+/// can be stored in its own partition. Keys belonging to an account are identified by [`KeyLocation`]s
+/// in that partition.
+/// An `IotaDID` can be resolved to an `AccountId` using the index, which is a global data structure.
+/// Therefore, index operations need to be carefully synchronized per [`Storage`] instance, while other operations
+/// only need to be synchronized within the same `AccountId`.
+/// For example, two `key_generate` operations with different account ids can be executed concurrently.
```

Comment:
This is actually incorrect, we don't expect `Storage` to synchronize per account. I will reword this.